helix-term/src/commands.rs

```diff
@@ -5115,11 +5115,22 @@ fn surround_replace(cx: &mut Context) {
             }
         };
 
+        // Visual feedback
+        let selection = selection.clone();
+        let mut ranges: SmallVec<[Range; 1]> = SmallVec::new();
+        // TODO: Use [`array_chunks`] once stabilized
+        change_pos.chunks_exact(2).for_each(|p| {
```

Comment:
`for` is considered more idiomatic than `Iterator::for_each` when not many methods are chained like in this case. See https://doc.rust-lang.org/core/iter/trait.Iterator.html#method.for_each

```rust
for point in change_pos.chunks_exact(2) {
    let [from, to] = *point else { unreachable!() };
    ranges.push(Range::point(from));
    ranges.push(Range::point(to));
}
```

Reply:
Learning a lot! Thx